File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/javaBuilder/axis3DAPIs/SciChart3DBasicsCoordinatesIn3DSpace.java
package com.scichart.docsandbox.examples.javaBuilder.axis3DAPIs;

import androidx.annotation.NonNull;

import com.scichart.charting.numerics.coordinateCalculators.ICoordinateCalculator;
import com.scichart.charting3d.visuals.SciChartSurface3D;
import com.scichart.charting3d.visuals.axes.NumericAxis3D;
import com.scichart.docsandbox.core.ExampleDefinition;
import com.scichart.docsandbox.examples.base.SingleChart3DFragment;

@ExampleDefinition()
public class SciChart3DBasicsCoordinatesIn3DSpace extends SingleChart3DFragment {
    @Override
    protected void initExample(@NonNull SciChartSurface3D surface) { }

    void setIsLeftHandedCoordinateSystem(@NonNull SciChartSurface3D surface) {
        // <SetIsLeftHandedCoordinateSystem>
        surface.getViewport3D().setIsLeftHandedCoordinateSystem(false);
        // </SetIsLeftHandedCoordinateSystem>
    }

    void setWorldCoordinates(@NonNull SciChartSurface3D surface) {
        // <SetWorldCoordinates>
        surface.getWorldDimensions().assign(200, 200, 200);
        // </SetWorldCoordinates>
    }

    void convertingFromWorldToDataCoordinates() {
        // <ConvertingFromWorldToDataCoordinates>
        final NumericAxis3D xAxis = new NumericAxis3D();

        ICoordinateCalculator calculator = xAxis.getCurrentCoordinateCalculator();
        float coordinate = calculator.getCoordinate(1.2);

        // Convert back:
        double dataValue = calculator.getDataValue(coordinate);
        // </ConvertingFromWorldToDataCoordinates>
    }
}

File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/kotlin/axis3DAPIs/SciChart3DBasicsCoordinatesIn3DSpace.kt
package com.scichart.docsandbox.examples.kotlin.axis3DAPIs

import com.scichart.charting3d.visuals.SciChartSurface3D
import com.scichart.charting3d.visuals.axes.NumericAxis3D
import com.scichart.docsandbox.core.ExampleDefinition
import com.scichart.docsandbox.examples.base.SingleChart3DFragment

@ExampleDefinition()
class SciChart3DBasicsCoordinatesIn3DSpace : SingleChart3DFragment() {
    override fun initExample(surface: SciChartSurface3D) {}

    fun setIsLeftHandedCoordinateSystem(surface: SciChartSurface3D) {
        // <SetIsLeftHandedCoordinateSystem>
        surface.viewport3D.isLeftHandedCoordinateSystem = false
        // </SetIsLeftHandedCoordinateSystem>
    }

    fun setWorldCoordinates(surface: SciChartSurface3D) {
        // <SetWorldCoordinates>
        surface.worldDimensions.assign(200f, 200f, 200f)
        // </SetWorldCoordinates>
    }

    fun convertingFromWorldToDataCoordinates() {
        // <ConvertingFromWorldToDataCoordinates>
        val xAxis = NumericAxis3D()
        val calculator = xAxis.currentCoordinateCalculator
        val coordinate = calculator.getCoordinate(1.2)

        // Convert back:
        val dataValue = calculator.getDataValue(coordinate)
        // </ConvertingFromWorldToDataCoordinates>
    }
}